src/java/org/apache/cassandra/cql3/statements/TruncateStatement.java

```diff
@@ -119,4 +140,43 @@ public AuditLogContext getAuditLogContext()
     {
         return new AuditLogContext(AuditLogEntryType.TRUNCATE, keyspace(), name());
     }
+
+    public static final class Raw extends QualifiedStatement<TruncateStatement>
+    {
+        public Raw(QualifiedName name)
+        {
+            super(name);
+        }
+
+        @Override
+        public TruncateStatement prepare(ClientState state, UnaryOperator<String> keyspaceMapper)
+        {
+            setKeyspace(state);
+            String ks = keyspaceMapper.apply(keyspace());
+            QualifiedName qual = qualifiedName;
+            if (!ks.equals(qual.getKeyspace()))
+                qual = new QualifiedName(ks, qual.getName());
+
+            // local keyspaces are always local so don't use the remote statement even if it has been configured
+            if (!REMOTE_TRUNCATE_STATEMENT.isPresent() ||
```

Comment:
well, I think the factoryClass should be obtained once - do we need to do that for each truncate statement execution?

Reply:
In all other places obtaining a class from the factory is done dynamically every time. The issue is that we don't know if it can be obtained once or not as it depends on how this extension is used. Do you have better suggestions?

Reply:
hmm...  system properties should not be treated as dynamic. To support testing this, I would do the following:
- instead of defining a statement class name in a system property, I'd define a provider class name
- the default provider and default remote providers would return value of a final field
- the test provider could return value of a static mutable field

This way we can avoid every time initialization while keeping the required flexibility for testing, wdyt?

Reply:
note that it would also allow returning a mock for testing which could be then easily verified if it was used

Reply:
@jacek-lewandowski Thank you for your suggestion. I will look at it.



> hmm... system properties should not be treated as dynamic.

I noticed I am not the first one who is doing it in the code base. Is it worth a ticket for future cleanup? For example, see https://github.com/datastax/cassandra/blame/ds-trunk/src/java/org/apache/cassandra/io/sstable/StorageHandler.java#L147

Reply:
This was the practice in Cassandra code and we somehow inherited it but I doesn't work well due to the reasons I mentioned above. It would be great to clean that up at some point, a ticket is a good idea. We should also consider creating the related changes in OSS in the future

Reply:
My link is to the code introduced in CC and it doesn't exist in OSS.

Reply:
I implemented the suggestion in my PR.